### PR TITLE
ci: Print stats on covered files modifed in diff

### PR DESCRIPTION
**Pull request by jamesclark-Zapata**
# The problem

Checking coverage is becoming more difficult as the repo gets larger.

# This PR's solution

Adds a section to the PR command that looks at the diff against the base branch and posts a diff using [`diff-cover`](https://pypi.org/project/diff-cover/).

# Checklist

_Check that this PR satisfies the following items:_

- [x] Tests have been added for new features/changed behavior (if no new features have been added, check the box).
- [x] The [changelog file](CHANGELOG.md) has been updated with a user-readable description of the changes (if the change isn't visible to the user in any way, check the box).
- [x] The PR's title is prefixed with `<feat/fix/chore/internal/docs>[!]:`

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -46,6 +46,11 @@ coverage:
 show-coverage-text-report:
 	$(PYTHON) -m coverage report --show-missing
 
+BASE_BRANCH := $(if $(GITHUB_BASE_REF),$(GITHUB_BASE_REF),main)
+github-actions-coverage-report:
+	@$(PYTHON) -m coverage report --show-missing
+	@$(PYTHON) -m diff_cover.diff_cover_tool coverage.xml --compare-branch=origin/$(BASE_BRANCH)
```
Comment (alexjuda) [inline Makefile:52]: **praise**: nice!!!

**Review by alexjuda — APPROVED** [review]
LGTM. Have you tried it on a PR with diffs in the source code?

**Review by BenjaminMummery — APPROVED** [review]
This is going to be really useful!

**Comment by alexjuda** [comment]
Moved to https://github.com/zapatacomputing/orquestra-workflow-sdk/pull/17

**Pull request closed without merging**